### PR TITLE
3655/download test client dependency upgrades

### PR DESCRIPTION
**Pull request by EVogel1999**
This PR updates various dependencies in the download test client and makes the switch to node-fetch.  Completes story [3655/download test client dependency upgrades](https://app.clubhouse.io/clarkcan/story/3655/download-test-client-dependency-upgrades).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,42 +10,33 @@
     "start-server": "nodemon",
     "serve": "node -r ts-node/register ./src/adapters/app.ts",
     "serve:debug": "node --inspect=5858 -r ts-node/register ./src/app.ts",
-    "build": "node_modules/.bin/gulp tsc",
-    "postinstall": "node_modules/.bin/gulp tsc",
-    "gulp-tsc": "node_modules/.bin/gulp start",
+    "build": "tsc",
```
Comment (pzalep1) [inline package.json:13]: I think this is the official death to gulp on our system!

**Review by pzalep1 — APPROVED** [review]
This looks good!

File: src/service-tests/download-tests.ts [hunk 2]
```diff
@@ -75,6 +75,22 @@ function setOptions(uri: string, token: string): void {
...
 
+/**
+ * This checks the status of the response from the node-fetch
+ * statement to see if it threw an error message
+ * 
+ * @param res The node-fetch response object
+ * @returns The json of the request or throws and error
+ */
+ function checkStatus(res: any): Promise<any> {
+    // res.status >= 200 && res.status < 300
```
Comment (Cwagne17) [inline src/service-tests/download-tests.ts:86]: How come it says 300 in the commented out and 600 in the conditional

**Review by Cwagne17 — APPROVED** [review]
Other then the build and the question it looks muy bueno

**Review by devky690 — APPROVED** [review]

**Review by mfranz1 — APPROVED** [review]

File: src/service-tests/download-tests.ts (hunk 2 above)
Comment (EVogel1999) [reply src/service-tests/download-tests.ts]: Forgot to remove this, this is what happens if you use res.ok.

**Pull request merged**